HUDI-162 : File System view must be built with correct timeline actions

bvaradar opened the pull request.

n3nash commented on hoodie-common/src/main/java/com/uber/hoodie/common/table/HoodieTimeline.java:
```diff
@@ -105,6 +105,12 @@
+   * Filter this timeline to just include commits (commit/deltacommit) and compaction actions
+   * @return
+   */
+  HoodieTimeline filterCommitsAndCompactionTimeline();
```
There is already a getCommitsAndCompactionTimeline in HoodieActiveTimeline, do we need this ?

vinothchandar replied:
+1

bvaradar replied:
I need similar method in HoodieTimeline interface. I have renamed the method to getCommitsAndCompactionTimeline so that they will be overridden by HoodieDefaultTimeline and HoodieActiveTImeline.

n3nash commented on hoodie-common/src/main/java/com/uber/hoodie/common/table/view/AbstractTableFileSystemView.java:
```diff
@@ -92,20 +92,29 @@ private String getPartitionPathFromFilePath(String fullPath) {
+   * Update commits timeline
+   * @param visibleActiveTimeline  Visible Active Timeline
+   */
+  protected void updateTimeline(HoodieTimeline visibleActiveTimeline) {
```
rename to getCommitsAndCompactionTimeline()..

vinothchandar replied:
refreshTimeline()?

bvaradar replied:
Done

vinothchandar commented on hoodie-common/src/main/java/com/uber/hoodie/common/table/view/AbstractTableFileSystemView.java:
```diff
@@ -822,10 +832,18 @@ public void sync() {
+   * Return Only Commits and Compaction timeline for building file-groups
+   * @return
+   */
+  public HoodieTimeline getVisibleCommitsAndCompactionTimeline() {
```
is this needed?  seems redundant with `getTimeline()`?

bvaradar replied:
This is needed to distinguish active timeline from commitsAndCompactionTimeline

vinothchandar commented on hoodie-common/src/main/java/com/uber/hoodie/common/table/view/AbstractTableFileSystemView.java:
```diff
@@ -216,7 +224,7 @@ private void ensurePartitionLoadedCorrectly(String partition) {
 
           // Create the path if it does not exist already
-          Path partitionPath = new Path(metaClient.getBasePath(), partitionPathStr);
+          Path partitionPath = FSUtils.getPartitionPath(metaClient.getBasePath(), partitionPathStr);
```
can you do one grep for all such occurrences across the code ?

bvaradar replied:
Fixed more occurrences of this pattern

bvaradar commented:
@n3nash @vinothchandar : Addressed/Replied-to review comments. Unit tests and automated hoodie-demo (manually triggered) ran fine. As discussed earlier, will go ahead merging this change.

@cdmikechen : I addressed the exception " java.lang.IllegalArgumentException: Can not create a Path from an empty string"  as part of this PR.  The pattern was found in few other places and I felt it is best handled together.
